spec/acceptance/conf_setting_spec.rb

```diff
@@ -57,23 +71,26 @@
       EOS
 
       it 'applies the manifest twice' do
-        idempotent_apply(pp)
+        apply_manifest(pp, catch_failures: true)
+        apply_manifest(pp, catch_changes: true)
       end
 
       describe file("#{tmpdir}/hocon_setting.conf") do
         it { is_expected.to be_file }
-        it("contains one {\n two=three\n}\nfour=five") {
+        # XXX Solaris 10 doesn't support multi-line grep
+
+        it("contains one {\n two=three\n}\nfour=five", unless: fact('osfamily') == 'Solaris') {
           is_expected.to contain("one {\n    two=three\n}\nfour=five")
         }
       end
     end
 
     context '=> absent for key/value' do
       before :all do
-        if os[:family] == 'Darwin'
-          run_shell("echo \"one {\n    two=three\n}\nfour=five\" > #{tmpdir}/hocon_setting.conf")
+        if fact('osfamily') == 'Darwin'
```

Comment:
`fact()` runs a shell command every time, while we don't even test on Darwin. Should we drop this?